packages/discovery-react-components/src/components/DiscoverySearch/DiscoverySearch.tsx

```diff
@@ -287,31 +289,22 @@ const DiscoverySearch: FC<DiscoverySearchProps> = ({
       backwardsCompatibleQueryParams: DiscoveryV2.QueryParams & { returnFields?: string[] },
       resetAggregations = true
     ): Promise<void> => {
-      let aggregationsFetched = false;
       const searchParameters = deprecateReturnFields(
         backwardsCompatibleQueryParams
       ) as DiscoveryV2.QueryParams;
       setSearchParameters(searchParameters);
-      // don't use the search response if filter is set, just do another search
-      if (resetAggregations && searchParameters.filter !== '') {
-        aggregationsFetched = true;
-        fetchGlobalAggregations(
-          {
-            ...searchParameters,
-            ...aggregationQueryDefaults,
-            filter: ''
-          },
-          async aggregations => {
-            fetchTypeForTopEntitiesAggregation(aggregations, searchParameters);
-          }
-        );
+
+      if (resetAggregations) {
```

Comment:
i think the proper fix is probably to just move any calls to `handleSearch` with `resetAggregations = true` should just call `fetchAggregations` instead

Reply:
Should we add a comment here?